app/Media/ImageStorageProvider.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Media;
+
+use Aws\S3\S3Client;
+use CultuurNet\UDB3\Container\AbstractServiceProvider;
+use League\Flysystem\AwsS3V3\AwsS3V3Adapter;
+use League\Flysystem\Filesystem;
+use League\Flysystem\Local\LocalFilesystemAdapter;
+
+final class ImageStorageProvider extends AbstractServiceProvider
+{
+    protected function getProvidedServiceNames(): array
+    {
+        return [
+            'local_file_system',
+            's3_file_system',
+            'image_storage',
+        ];
+    }
+
+    public function register(): void
+    {
+        $container = $this->getContainer();
+
+        $container->addShared(
+            'local_file_system',
+            function (): Filesystem {
+                $localAdapter = new LocalFilesystemAdapter(__DIR__ . '/../../');
+                return new Filesystem($localAdapter);
+            }
+        );
+
+        $container->addShared(
+            's3_file_system',
```

Comment:
Nit: This is only used in the `image_storage` so you could move that inline already, but can also be done later